src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -749,6 +748,14 @@ public static void onEmptyLeftClick(Player player)
     {
         MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.LeftClickEmpty(player));
     }
+    
+    public static InteractionResult onItemUse(ItemStack stack, UseOnContext ctx)
+    {
+        PlayerInteractEvent.ItemUseEvent event = new PlayerInteractEvent.ItemUseEvent(ctx);
+        MinecraftForge.EVENT_BUS.post(event);
+        if (event.isCanceled()) return event.getCancellationResult();
```

Comment:
Because this effect is affected by calls to `PlayerInteractEvent#setCancellationResult`, the javadoc for that method should be amended to include this new event in its list of affected events.